core/src/proc_macro/context.rs

```diff
@@ -40,7 +41,8 @@ impl Arguments {
 
     /// Parses the JSON representation from CARGO_LIGEN_ARGUMENTS.
     pub fn from_env() -> Self {
-        let json_string = std::env::var("CARGO_LIGEN_ARGUMENTS").expect("Couldn't set the environment variables.");
+        let json_string = std::env::var("CARGO_LIGEN_ARGUMENTS")
+            .expect("Couldn't set the environment variables.");
         serde_json::from_str(&json_string).expect("Couldn't parse JSON string.")
     }
 }
```

Comment:
Could you turn `-> Self` into `-> Result<Self, Error>`?

Reply:
Can it be `-> Result<Self, String>`?